src/ngraph/op/scatter_elements_update.cpp

```diff
@@ -0,0 +1,84 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/scatter_elements_update.hpp"
+
+using namespace ngraph;
+using namespace std;
+
+constexpr NodeTypeInfo op::v0::ScatterElementsUpdate::type_info;
+
+op::v0::ScatterElementsUpdate::ScatterElementsUpdate(const Output<Node>& data,
+                                                     const Output<Node>& indices,
+                                                     const Output<Node>& updates,
+                                                     const Output<Node>& axis)
+    : Op({data, indices, updates, axis})
+{
+    constructor_validate_and_infer_types();
+}
+
+bool ngraph::op::v0::ScatterElementsUpdate::visit_attributes(AttributeVisitor& visitor)
+{
+    return true;
+}
+
+void op::v0::ScatterElementsUpdate::validate_and_infer_types()
+{
+    element::Type data_et = get_input_element_type(0);
+    element::Type indices_et = get_input_element_type(1);
+    element::Type updates_et = get_input_element_type(2);
+    element::Type axis_et = get_input_element_type(3);
+
+    const PartialShape& data_shape = get_input_partial_shape(0);
+    const PartialShape& indices_shape = get_input_partial_shape(1);
+    const PartialShape& updates_shape = get_input_partial_shape(2);
+    const PartialShape& axis_shape = get_input_partial_shape(3);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_et == element::i32 || indices_et == element::i64,
```

Comment:
I think that we can relax constraints to `indices_et.is_integral_number()'. I know that it is requirement from spec, but for other ops from `opset1` we have such attitude (https://github.com/NervanaSystems/ngraph/pull/4191)